packages/salesforcedx-sobjects-faux-generator/src/transformer/transformerFactory.ts

```diff
@@ -87,8 +90,24 @@ export class SObjectTransformerFactory {
   }
 
   public static async createConnection(): Promise<Connection> {
-    const connection = await WorkspaceContextUtil.getInstance().getConnection();
-    return connection;
+    const userApiVersionOverride = await ConfigUtil.getUserConfiguredApiVersion();
+    const workspaceContextUtil = WorkspaceContextUtil.getInstance();
+    const connection = await workspaceContextUtil.getConnection();
+    const connectionForSourceApiVersion = await Connection.create({
+      authInfo: await AuthInfo.create({ username: connection.getUsername() })
+    });
+    try {
+      const sfProject = await SfProject.resolve();
+      const sourceApiVersion = sfProject.getSfProjectJson().getContents()
+        .sourceApiVersion;
+      // precedence user override > project config > connection default
+      connectionForSourceApiVersion.setApiVersion(
+        userApiVersionOverride || sourceApiVersion || connection.getApiVersion()
+      );
+    } catch (e) {
+      // If we can't resolve a project, we'll just use the default connection
```

Comment:
We like to avoid empty catch blocks.  When you were implementing this, was there something that was throwing?  What were the repro steps?